### PR TITLE
Add optional Pod Disruption Budget for each potential Pega tier

### PR DESCRIPTION
**Pull request by RyanStan**
By default, these changes won't have an impact on client managed cloud unless specifically specified.  Customers can either specify the minimum number of available pods they want available at all times for a specific tier, or the maximum number of pods that can be unavailable at one time.  

For CloudK customers, the Pod Disruption Budget will be added by default with parameters set so that one pod in each tier must be available at all times.

### CODE REVIEW

**Review by kishorv10 — COMMENTED** [review]
Can you add helm template test cases for these changes?

**Review by APegaDavis — APPROVED** [review]

File: charts/pega/README.md [hunk 1]
```diff
@@ -480,6 +480,17 @@ Parameter           | Description    | Default value
 `hpa.enableCpuTarget` | Set to true if you want to enable scaling based on CPU utilization or false if you want to disable it | true
 `hpa.enableMemoryTarget` | Set to true if you want to enable scaling based on memory utilization or false if you want to disable it (Pega recommends leaving this disabled) | false
 
+### Setting a Pod Disruption Budget (PDB)
+You can configure a Kubernetes `PodDisruptionBudget`on your tier by setting `pdb.enabled` to `true`.  For more details, see the Kubernetes
```
Comment (pega-talba) [inline charts/pega/README.md:484]:
```suggestion
You can configure a Kubernetes `PodDisruptionBudget` on your tier by setting `pdb.enabled` to `true`.  For more details, see the Kubernetes 
```

File: charts/pega/README.md [hunk 2]
```diff
@@ -480,6 +480,17 @@ Parameter           | Description    | Default value
 `hpa.enableCpuTarget` | Set to true if you want to enable scaling based on CPU utilization or false if you want to disable it | true
 `hpa.enableMemoryTarget` | Set to true if you want to enable scaling based on memory utilization or false if you want to disable it (Pega recommends leaving this disabled) | false
 
+### Setting a Pod Disruption Budget (PDB)
+You can configure a Kubernetes `PodDisruptionBudget`on your tier by setting `pdb.enabled` to `true`.  For more details, see the Kubernetes 
+[Pod Disruption Budgets documentation](https://kubernetes.io/docs/concepts/workloads/pods/disruptions/#pod-disruption-budgets).
+
+The following parameters are mutually exclusive.  You may specifiy either `pdb.minAvailable` or `pdb.maxUnavailable` but not both.
```
Comment (pega-talba) [inline charts/pega/README.md:487]:
```suggestion
The following parameters are mutually exclusive.  You must specify either `pdb.minAvailable` or `pdb.maxUnavailable` but not both.
```

**Review by misterdorito — APPROVED** [review]

**Comment by MadhuriArugula** [comment]
@taz-pega-work Can you review the documentation changes in this PR to proceed with merge?

**Review by kishorv10 — APPROVED** [review]

File: charts/pega/README.md [hunk 3]
```diff
@@ -480,6 +480,17 @@ Parameter           | Description    | Default value
 `hpa.enableCpuTarget` | Set to true if you want to enable scaling based on CPU utilization or false if you want to disable it | true
 `hpa.enableMemoryTarget` | Set to true if you want to enable scaling based on memory utilization or false if you want to disable it (Pega recommends leaving this disabled) | false
 
+### Setting a Pod Disruption Budget (PDB)
+You can configure a Kubernetes `PodDisruptionBudget` on your tier by setting `pdb.enabled` to `true`.  For more details, see the Kubernetes
```
Comment (taz-pega-work) [inline charts/pega/README.md:484]: for line 483, can we add a context and not rely on the user to put together performance and a "disruption budget" as an equivalence?
To safeguard your deployment by limiting the number of Pods running your Pega Platform application that can go down for planned "disruptions", Pega allows you to enable a Kubernetes `PodDisruptionBudget`. For more details, see the Kubernetes...

File: charts/pega/README.md [hunk 4]
```diff
@@ -480,6 +480,17 @@ Parameter           | Description    | Default value
...
+### Setting a Pod Disruption Budget (PDB)
+You can configure a Kubernetes `PodDisruptionBudget` on your tier by setting `pdb.enabled` to `true`.  For more details, see the Kubernetes 
+[Pod Disruption Budgets documentation](https://kubernetes.io/docs/concepts/workloads/pods/disruptions/#pod-disruption-budgets).
+
+The following parameters are mutually exclusive.  You must specify either `pdb.minAvailable` or `pdb.maxUnavailable` but not both.
+
+Parameter             | Description    | Default value
+---                   | ---       | ---
+`pdb.minAvailable`    | The minimum number of pods in the tier that must remain available at all times  | `1`
+`pdb.maxUnavailable`  | The maximum number of pods in the tier that can be unavailable at one time      | Disabled when `pdb.minAvailable` is specified
```
Comment (taz-pega-work) [inline charts/pega/README.md:492]: this isn't a value, it's information about usage. Should this be in the description? also, make it active>>Disabled when you specify a `pdb.minAvailable` value.

Should this mention percentage if the commented out values are percentages?

File: charts/pega/README.md [hunk 5]
```diff
@@ -480,6 +480,17 @@ Parameter           | Description    | Default value
 `hpa.enableCpuTarget` | Set to true if you want to enable scaling based on CPU utilization or false if you want to disable it | true
 `hpa.enableMemoryTarget` | Set to true if you want to enable scaling based on memory utilization or false if you want to disable it (Pega recommends leaving this disabled) | false
 
+### Setting a Pod Disruption Budget (PDB)
+You can configure a Kubernetes `PodDisruptionBudget` on your tier by setting `pdb.enabled` to `true`.  For more details, see the Kubernetes 
+[Pod Disruption Budgets documentation](https://kubernetes.io/docs/concepts/workloads/pods/disruptions/#pod-disruption-budgets).
+
+The following parameters are mutually exclusive.  You must specify either `pdb.minAvailable` or `pdb.maxUnavailable` but not both.
```
Comment (taz-pega-work) [inline charts/pega/README.md:487]: I think you should state that clients configure this in their deployment values.yaml file and it's disabled ('false') by default. Also, mention that you must use comments to disable the other setting.

Finally, the values.yaml files each have a 50% value commented out, not an integer so I'm automatically confused.

File: charts/pega/values-large.yaml [hunk 6]
```diff
@@ -125,6 +125,12 @@ global:
       hpa:
         enabled: true
 
+      # Set enabled to true to include a Pod Disruption Budget for this tier
```
Comment (taz-pega-work) [inline charts/pega/values-large.yaml:128]: Set enabled to true to include a Pod Disruption Budget for this tier. To enable this budget, specify either a `pdb.minAvailable` or `pdb.maxUnavailable` value and comment out the other parameter.

Replicate this wording throughout the values.yaml files. Right now, I think it's a bit too un-documented on what I have to do to make it work for maxUnavailable.

**Review by taz-pega-work — APPROVED** [review]
I hope you'll consider clarifying a few items in the files so it's more clear on how to use maxUnavailable.

File: charts/pega/README.md [hunk 7]
```diff
@@ -480,6 +480,19 @@ Parameter           | Description    | Default value
 `hpa.enableCpuTarget` | Set to true if you want to enable scaling based on CPU utilization or false if you want to disable it | true
 `hpa.enableMemoryTarget` | Set to true if you want to enable scaling based on memory utilization or false if you want to disable it (Pega recommends leaving this disabled) | false
 
+### Ensure System Availability during Voluntary Disruptions by Using a Kubernetes Pod Disruption Budget (PDB)
+To safeguard your deployment by limiting the number of Pods running your Pega Platform application that can go down for planned "disruptions", 
+Pega allows you to enable a unique Kubernetes `PodDisruptionBudget` on a tier.  For more details on PDBs, see the Kubernetes [Pod Disruption Budgets documentation](https://kubernetes.io/docs/concepts/workloads/pods/disruptions/#pod-disruption-budgets).
+
+You can configure a Kubernetes `PodDisruptionBudget` on your tier by setting `pdb.enabled` to `true` in your values.yaml file.  By default, this value is
+set to `false.`  You must also specify at least one of the following parameters to configure the Pod Disruption Budget.  Keep in mind that these parameters
```
Comment (pega-talba) [inline charts/pega/README.md:488]:
```suggestion
set to `false.`  You must also specify exactly one of the following parameters to configure the Pod Disruption Budget.  These parameters
```

File: charts/pega/README.md [hunk 8]
```diff
@@ -480,6 +480,19 @@ Parameter           | Description    | Default value
 `hpa.enableCpuTarget` | Set to true if you want to enable scaling based on CPU utilization or false if you want to disable it | true
 `hpa.enableMemoryTarget` | Set to true if you want to enable scaling based on memory utilization or false if you want to disable it (Pega recommends leaving this disabled) | false
 
+### Ensure System Availability during Voluntary Disruptions by Using a Kubernetes Pod Disruption Budget (PDB)
+To safeguard your deployment by limiting the number of Pods running your Pega Platform application that can go down for planned "disruptions", 
+Pega allows you to enable a unique Kubernetes `PodDisruptionBudget` on a tier.  For more details on PDBs, see the Kubernetes [Pod Disruption Budgets documentation](https://kubernetes.io/docs/concepts/workloads/pods/disruptions/#pod-disruption-budgets).
```
Comment (pega-talba) [inline charts/pega/README.md:485]:
```suggestion
Pega allows you to enable a Kubernetes `PodDisruptionBudget` on a tier.  For more details on PDBs, see the Kubernetes [Pod Disruption Budgets documentation](https://kubernetes.io/docs/concepts/workloads/pods/disruptions/#pod-disruption-budgets).
```

File: charts/pega/README.md [hunk 9]
```diff
@@ -480,6 +480,19 @@ Parameter           | Description    | Default value
...
+
+You can configure a Kubernetes `PodDisruptionBudget` on your tier by setting `pdb.enabled` to `true` in your values.yaml file.  By default, this value is
+set to `false.`  You must also specify at least one of the following parameters to configure the Pod Disruption Budget.  Keep in mind that these parameters
+are mutually exclusive, and thus only one may be set.  You may provide values that are expressed as percentages (% of pods) or integers (# of pods).  
+
+Parameter             | Description    | Default value
+---                   | ---       | ---
+`pdb.minAvailable`    | The minimum number of pods in the tier that must remain available at all times  | `1`
+`pdb.maxUnavailable`  | The maximum number of pods in the tier that can be unavailable at one time      | `50%`
```
Comment (pega-talba) [inline charts/pega/README.md:494]:
```suggestion
`pdb.maxUnavailable`  | The maximum number of pods in the tier that can be unavailable at one time      | `50%` (disabled by default)
```

File: charts/pega/README.md [hunk 10]
```diff
@@ -480,6 +480,19 @@ Parameter           | Description    | Default value
 `hpa.enableCpuTarget` | Set to true if you want to enable scaling based on CPU utilization or false if you want to disable it | true
 `hpa.enableMemoryTarget` | Set to true if you want to enable scaling based on memory utilization or false if you want to disable it (Pega recommends leaving this disabled) | false
 
+### Ensure System Availability during Voluntary Disruptions by Using a Kubernetes Pod Disruption Budget (PDB)
+To safeguard your deployment by limiting the number of Pods running your Pega Platform application that can go down for planned "disruptions",
```
Comment (pega-talba) [inline charts/pega/README.md:484]:
```suggestion
To limit the number of Pods running your Pega Platform application that can go down for planned disruptions, 
```

File: charts/pega/README.md [hunk 11]
```diff
@@ -480,16 +480,18 @@ Parameter           | Description    | Default value
...
-The following parameters are mutually exclusive.  You must specify either `pdb.minAvailable` or `pdb.maxUnavailable` but not both.
+You can configure a Kubernetes `PodDisruptionBudget` on your tier by setting `pdb.enabled` to `true` in your values.yaml file.  By default, this value is
+set to `false.`  You must also specify at least one of the following parameters to configure the Pod Disruption Budget.  Keep in mind that these parameters
+are mutually exclusive, and thus only one may be set.  You may provide values that are expressed as percentages (% of pods) or integers (# of pods).  
 
 Parameter             | Description    | Default value
 ---                   | ---       | ---
 `pdb.minAvailable`    | The minimum number of pods in the tier that must remain available at all times  | `1`
-`pdb.maxUnavailable`  | The maximum number of pods in the tier that can be unavailable at one time      | Disabled when `pdb.minAvailable` is specified
+`pdb.maxUnavailable`  | The maximum number of pods in the tier that can be unavailable at one time      | `50%`
```
Comment (taz-pega-work) [inline charts/pega/README.md]: sorry but I think that this is still off...how about?>
The minumum number or percentage of total pods in the tier that can be unavailable at one time; if the number or percentage reaches your minimum, the deployment will not allow you to bring down additional pods until one of the currently unavailable pods is deployed and healthy.

The maximum number or percentage of total pods in the tier that can be unavailable at one time; if the number or percentage reaches your maximum, the deployment will not allow you to bring down additional pods until one of the currently unavailable pods is deployed and healthy.

**Review by taz-pega-work — APPROVED** [review]
looks really good...one little thing let to suss out. thanks and great job clarifying the statements!

File: charts/pega/README.md (hunk 11 above)
Comment (pega-talba) [reply charts/pega/README.md]: I like the inclusion of percentage, but the suggested description isn't quite technically accurate. Maybe something like this?

> The minimum number or percentage of pods in the tier that must be available. If this minimum is reached, the Kubernetes deployment will not bring down additional pods for voluntary disruptions until more are available and healthy.

> The maximum number or percentage of pods in the tier that can be unavailable. If this maximum is reached, the Kubernetes deployment will not bring down additional pods for voluntary disruptions until more are available and healthy.

**Pull request merged**